src/lib/logical_query_plan/abstract_non_query_node.cpp

```diff
@@ -12,4 +12,12 @@ bool AbstractNonQueryNode::is_column_nullable(const ColumnID column_id) const {
   Fail("Node does not return any column");
 }
 
+std::shared_ptr<LQPUniqueConstraints> AbstractNonQueryNode::unique_constraints() const {
+  Fail("Node does not support unique constraints.");
+}
+
+std::vector<FunctionalDependency> AbstractNonQueryNode::functional_dependencies() const {
+  Fail("Node does not support functional dependencies.");
+}
```

Comment:
The above change led to errors in some tests. As it turns out, the corresponding LQPs involve `StaticTableNode`, a subclass of `AbstractNonQueryNode`. 

The optimizer is called during `TEST_P(MultiMetaTablesTest, SQLFeatures)`. Therefore, the Dependent-Group-By-Reduction-Rule fails when calling `functional_dependencies()`.

I fixed the issue by adding 
```cpp
std::vector<FunctionalDependency> StaticTableNode::functional_dependencies() const {
  // We could construct some FDs here, similar to StoredTableNode.
  // We did not do so yet because StaticTableNode is rarely used.
  return {};
}
```

To me, StaticTableNode seems to be only exception, or am I wrong? If there are more exceptions, I would remove the above Fails for `unique_constraints()` and `functional_dependencies()`.

Reply:
Mmh, why is StaticTableNode a AbstractNonQueryNode in the first place?

Reply:
I guess it was marked as such because of the following doc:

> This node type wraps a table and can be used as input for a CreateTableNode to represent a simple CREATE TABLE management command.

Seems that AbstractNonQueryNode did not receive much love. Other nodes such as InsertNode or DeleteNode should inherit, but they do not. Time to clean things up.